src/market/market.module.ts

```diff
@@ -625,4 +611,33 @@ export class MarketModuleImpl implements MarketModule {
       }
     });
   }
+
+  private filterProposalsByUserFilter(filter: ProposalFilter, proposal: OfferProposal) {
+    try {
+      const result = filter(proposal);
+
+      if (!result) {
+        this.events.emit("offerProposalRejectedByFilter", proposal);
+      }
+
+      return result;
+    } catch (err) {
+      this.logger.error("Executing user provided proposal filter resulted with an error", err);
+      throw err;
+    }
+  }
+
+  private filterProposalsByPricingOptions(pricing: PricingOptions, proposal: OfferProposal) {
+    let isPriceValid = true;
+    if (pricing.model === "linear") {
+      isPriceValid =
+        proposal.pricing.cpuSec <= pricing.maxCpuPerHourPrice / 3600 &&
+        proposal.pricing.envSec <= pricing.maxEnvPerHourPrice / 3600 &&
+        proposal.pricing.start <= pricing.maxStartPrice;
+    }
```

Comment:
what about `burn-rate`? We can check if start price + envSec * 3600 + cpuSec * 3600 is less than avgGlmPerHour